src/test/java/org/apache/commons/lang3/StringUtilsTest.java

```diff
@@ -3338,6 +3341,95 @@ public void testToRootUpperCase() {
         }
     }
 
+    @Test
+    void extractNumbersFromString() {
```

Comment:
By convention test method have the "test" prefix.

You should break up each use case in their own method IMO.

There are no tests for edge cases: null and empty string. Also a non empty string with no numbers.

Reply:
Valid points, I will add some. Thanks!